scormxblock/scormxblock.py

```diff
@@ -495,9 +495,13 @@ def _publish_grade(self, status, score):
 
     def publish_progress(self, scorm_data):
         """
-        Mark 100% completion if course is complete
+        Update progress % if cmi.progress_measure is emitted
+        Else check status and mark 100% completion if course is complete
         """
-        if scorm_data.get('status', '') in constants.SCORM_COMPLETION_STATUS:
+        progress_measure = self.calculate_progress_measure(scorm_data)
+        if progress_measure:
+            self._publish_progress(progress_measure)
+        elif scorm_data.get('status', '') in constants.SCORM_COMPLETION_STATUS:
```

Comment:
I think we don't need empty string here. We can change it to
`elif scorm_data.get('status') in constants.SCORM_COMPLETION_STATUS`

Reply:
Its just a good practice to not to use get alone, nothing trivial though.